### DISCUSSION TITLE
observe menu bar status

### DISCUSSION THREAD

**Discussion by castdrian** [orchetect/MenuBarExtraAccess#7] [Ideas]
so since the whole scene status doesn't work with the menubar, I really need a way to detect if the menubar menu is open or not since I need to periodically refresh some data based on that state, binding it to a bool to open it programmatically unfortunately doesn't affect the bool when the menu is opened manually :(

**Comment by orchetect** [comment]
Are you using a menu-based MenuBarExtra or window-based?

There is a limitation where the SwiftUI runloop is suspended when a traditional menu is open so the Bool binding may not update. That's a known issue that's documented in the README and in the demo project. Not sure if there's a workaround - Apple needs to address that issue but who knows if or when they will.

**Comment by castdrian** [comment]
eh don't worry about it, I settled with just refreshing the data in an interval instead
